session/cookie.js

```diff
@@ -7,146 +7,183 @@
+    const { expires, originalMaxAge, ...options } = data
+    const cookie = new Cookie(options)
+    cookie.expires = expires ? new Date(expires) : undefined
+    cookie.originalMaxAge = originalMaxAge
+    return cookie
   }
-};
-
-/*!
- * Prototype.
- */
-
-Cookie.prototype = {
 
   /**
    * Set expires `date`.
    *
-   * @param {Date} date
-   * @api public
+   * @param {Date | null | undefined} date
+   * @public
    */
 
   set expires(date) {
-    this._expires = date;
-    this.originalMaxAge = this.maxAge;
-  },
+    this._expires = date || undefined
+    this.originalMaxAge = this.maxAge
+  }
 
   /**
-   * Get expires `date`.
+   * Get expires `Date` object to be the value for the `Expires Set-Cookie` attribute.
+   * By default, no expiration is set, and most clients will consider this a "non-persistent cookie" and will delete it on a condition like exiting a web browser application.
    *
-   * @return {Date}
-   * @api public
+   * @return {Date | undefined}
+   * @public
    */
 
   get expires() {
-    return this._expires;
-  },
+    return this._expires
+  }
 
   /**
    * Set expires via max-age in `ms`.
    *
-   * @param {Number} ms
-   * @api public
+   * @param {number | undefined} ms
+   * @public
    */
 
   set maxAge(ms) {
-    if (ms && typeof ms !== 'number' && !(ms instanceof Date)) {
-      throw new TypeError('maxAge must be a number or Date')
-    }
-
-    if (ms instanceof Date) {
-      deprecate('maxAge as Date; pass number of milliseconds instead')
+    if (ms !== undefined) {
+      if (typeof ms !== 'number') {
+        throw new TypeError('maxAge must be a number')
+      }
+      this.expires = new Date(Date.now() + ms)
+    } else {
+      this.expires = undefined
     }
-
-    this.expires = typeof ms === 'number'
-      ? new Date(Date.now() + ms)
-      : ms;
-  },
+  }
 
   /**
    * Get expires max-age in `ms`.
    *
-   * @return {Number}
-   * @api public
+   * @return {number | undefined}
+   * @public
    */
 
   get maxAge() {
     return this.expires instanceof Date
       ? this.expires.valueOf() - Date.now()
-      : this.expires;
-  },
+      : this.expires
+  }
 
   /**
    * Return cookie data object.
    *
-   * @return {Object}
-   * @api private
+   * @return {cookie.CookieSerializeOptions}
+   * @private
    */
 
   get data() {
+    if (this.secure === 'auto') {
+      throw new Error("Invalid runtime state, the Cookie.secure == 'auto', which should not be possible.")
+    }
     return {
-      originalMaxAge: this.originalMaxAge,
       partitioned: this.partitioned,
       priority: this.priority
-      , expires: this._expires
+      , expires: this.expires
       , secure: this.secure
       , httpOnly: this.httpOnly
       , domain: this.domain
       , path: this.path
       , sameSite: this.sameSite
     }
-  },
-
-  /**
-   * Return a serialized cookie string.
-   *
-   * @return {String}
-   * @api public
-   */
-
-  serialize: function(name, val){
```

Comment:
I don't know if this is used by external Store implementations. In this repo, this was used only by the test code. The version in `index.js` calls `cookie.serialize` directly on purpose or by accident.

Reply:
First commit removed this method, but the second is more about just adding types and converting to a class, hence this method is back. The above comment still applies, and this method probably could be removed.